From 025dbb0e599ef1a00661371dd2bf90a3bfbb5abf Mon Sep 17 00:00:00 2001
From: jagaur <jagaur@habana.ai>
Date: Fri, 06 Jun 2025 19:08:43 +0300
Subject: [PATCH] [SW-230770] gaudi3: avoid /tmp dir contestation

This patch avoid possible contestation for common
directory during jenkins jobs run by creating temp dir
using mktemp.

Change-Id: I12aebfdedbd5a2aea14213d9d65e47304bba004b
---

diff --git a/board/habanalabs/common/post-image-spi-arc_global.sh b/board/habanalabs/common/post-image-spi-arc_global.sh
index b63db4e..ecc3a88 100755
--- a/board/habanalabs/common/post-image-spi-arc_global.sh
+++ b/board/habanalabs/common/post-image-spi-arc_global.sh
@@ -141,9 +141,10 @@
 	hl_reverse_words ${FILE} ${FILE}.out 4
 
 	# workround to avoid crc calculation error when absolute file path contains hex string
-	cp "${FILE}.out" /tmp/.
-	CRC32=$(crc32 /tmp/$(basename "${FILE}.out"))
-	rm -rf /tmp/$(basename "${FILE}.out")
+	TMPDIR=$(mktemp -d)
+	cp "${FILE}.out" "${TMPDIR}"/.
+	CRC32=$(crc32 "${TMPDIR}"/$(basename "${FILE}.out"))
+	rm -rf "${TMPDIR}"/
 
 	echo ${CRC32} | xxd -r -p > ${FILE}.crc32
 }
